Plugins/ch.zhaw.graphcut/src/internal/lib/GraphCut3D/ImageMultiLabelGridCutFilter.hxx

```diff
@@ -93,37 +91,43 @@ namespace itk {
 		unsigned int nGraphNodes(1);
 
         unsigned int dim(graphSize.GetSizeDimension());
-		for (int iSize = 0; iSize < dim; ++iSize) {
+		for (unsigned int iSize = 0; iSize < dim; ++iSize) {
 			nGraphNodes *= graphSize[iSize];
 		}
 
         WeightType * dataCosts = new WeightType[nGraphNodes * nLabels];
-        for (int iDatacost = 0; iDatacost < nGraphNodes * nLabels; ++iDatacost) {
-            if (std::numeric_limits<WeightType>::max()<std::numeric_limits<float>::max())
-                dataCosts[iDatacost] = std::numeric_limits<WeightType >::max() - 1;
-            else
-                dataCosts[iDatacost] = 1000;
+
+        WeightType dataCostTmp;
+        if (std::numeric_limits<WeightType>::max()<std::numeric_limits<float>::max())
+            dataCostTmp = std::numeric_limits<WeightType >::max() - 1;
+        else
+            dataCostTmp = 1000;
```

Comment:
I wanted here to use the type which would require the lowest size but I also wanted to be accurate enough. Then, the trick, is to spread the smooth costs on the entire range of the values which can be accepted by the type. Actually, I've decided to keep the smooth cost as an unsigned char but if we change our mind and we finaly decide to set it as a float we don't need to multiply by `std::numeric_limits<float>::max`, `1000` will be enough.